orca-peering/src/main/kotlin/com/netflix/spinnaker/orca/peering/ExecutionCopier.kt

```diff
@@ -152,13 +154,35 @@ open class ExecutionCopier(
         destDB.loadRecords(getStagesTable(executionType).name, stageRows)
       }
 
+      val partitionField = DSL.field("partition")
+
       // Step 2: Copy all executions
-      executionRows.forEach { r -> r.set(DSL.field("partition"), peeredId)
-        latestUpdatedAt = max(latestUpdatedAt, r.get("updated_at", Long::class.java))
+      if (executionRows.size < idsToMigrate.size) {
+        // There must've been some executions that have been deleted during our diff calculation, find out which ones
+        val idsToMigrateHash = idsToMigrate.toHashSet()
+
+        executionRows.forEach { r ->
+          r.set(partitionField, peeredId)
+          latestUpdatedAt = max(latestUpdatedAt, r.get("updated_at", Long::class.java))
+          idsToMigrateHash.remove(r.get("id"))
+        }
+
+        executionsNoLongerPresent.addAll(idsToMigrateHash)
+      } else {
+        executionRows.forEach { r ->
+          r.set(partitionField, peeredId)
+          latestUpdatedAt = max(latestUpdatedAt, r.get("updated_at", Long::class.java))
+        }
       }
       destDB.loadRecords(getExecutionTable(executionType).name, executionRows)
       peeringMetrics.incrementNumPeered(executionType, state, idsToMigrate.size)
 
+      // If we found any executions that are no longer there, delete them from the destination
+      if (executionsNoLongerPresent.any()) {
+        destDB.deleteExecutions(executionType, idsToMigrate)
```

Comment:
doesn't this delete all `idsToMigrate`, not just the ones in `executionsNoLongerPresent`?